src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -371,6 +371,18 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
             subscription=get_subscription_id(cmd.cli_ctx), resource_group=resource_group_name,
             namespace='Microsoft.Compute', type='diskAccesses', name=disk_access)
 
+    if security_type is not None:
+        if security_type.lower()[:15] == 'confidentialvm_' and not hyper_v_generation.lower() == 'v2':
```

Comment:
`hyper_v_generation` wouldn't it be `None` here?  Can users not pass in `hyper_v_generation` in this case?

Reply:
Got it, thanks~